### PR TITLE
Update for Flarum v1

### PR DESCRIPTION
**Pull request by dsevillamartin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ A [Flarum](http://flarum.org) extension. Merge two or more discussions into one.
 Use [Bazaar](https://discuss.flarum.org/d/5151-flagrow-bazaar-the-extension-marketplace) or install manually with composer:
```
Comment (imorland) [inline README.md]: We should remove the Bazaar reference
```suggestion
Install with composer:
```

**Review by imorland — COMMENTED** [review]
`category` in `composer.json` should also be changed to `feature`

Just my opinion, but we should probably also update the uses of `app()` to `resolve`. I know `app` is still available, but as it's permamently deprecated, we should lead by example and use the `resolve` method, imo

**Comment by imorland** [comment]
- After merging a previously split discussion, the modal remains open, and this is added to the destination discussion
 image

- Using `merge into me`, the merge modal remains open after the merge completes, rather than returning to the discussion
- Merged discussions are not removed from the discussion list after successful merge, if merged from within a discussion

**Comment by dsevillamartin** [comment]
> After merging a previously split discussion, the modal remains open, and this is added to the destination discussion

This will happen regardless of if split was used or not - e.g. if the issue with modal not closing happens and you click merge multiple times.

> Using merge into me, the merge modal remains open after the merge completes, rather than returning to the discussion

Had this happen but thought it was because I had changed something in the JS - I see now that was not the issue.

**Comment by dsevillamartin** [comment]
> Merged discussions are not removed from the discussion list after successful merge, if merged from within a discussion

I could not reproduce this.

I think I have fixed the other errors I found, however.

**Comment by Ralkage** [comment]
Just tested and haven't experienced any of the issues mentioned even after the recent changes @datitisev made; I believe we are good to merge this.

**Review by Ralkage — APPROVED** [review]

File: js/src/forum/components/DiscussionMergeModal.js [hunk 2]
```diff
@@ -179,27 +175,32 @@ export default class DiscussionMergeModal extends Modal {
         return app
             .request(this.getRequestData())
             .then(async () => {
+                const final = this.type() === 'target' ? this.discussion : this.merging[0];
+
                 if (app.current.matches(DiscussionPage)) {
+                    console.log('updating discussion page');
```
Comment (imorland) [inline js/src/forum/components/DiscussionMergeModal.js]:
```suggestion
```

File: js/src/forum/components/DiscussionMergeModal.js [hunk 3]
```diff
@@ -179,27 +175,32 @@ export default class DiscussionMergeModal extends Modal {
...
+                        await app.current.get('stream').update();
                     } else {
-                        m.route.set(app.route.discussion(this.merging[0]));
+                        m.route.set(app.route.discussion(final));
                     }
+                } else if (app.current.matches(IndexPage)) {
+                    await app.store.find('discussions', final.id());
                 }
 
+                console.log('removing old discussions');
```
Comment (imorland) [inline js/src/forum/components/DiscussionMergeModal.js]:
```suggestion
```

**Review by imorland — CHANGES_REQUESTED** [review]
On the whole this is much better now, nice one 👍 

I'll open a seperate issue for the `best-answer` side of things, at least so it does not get forgotten.

**Review by imorland — APPROVED** [review]

**Comment by dsevillamartin** [comment]
Nooo by debug console.log()s have leaked 😂

**Pull request merged**